DB/api/url_parser.py

```diff
@@ -22,16 +22,31 @@ def parse(url):
 		"reverse": False,
 		"device": None,
 		"diff": False,
-		}
+		"total_energy": False,
+		"energy_list": False,
+		"granularity": None
+	}
 
 	''' Extract device_id from URL '''
-	if re.match('^/\d+$', path):
-		query_options['device_id'] = int(path[1:])
-	else:
-		raise Exception("Serial Not Found")
+	device_id_only = re.match('^/(?P<device_id>\d+$)', path)
+	if device_id_only:
+		query_options['device_id'] = int(device_id_only.group('device_id'))
+
+	'''Extract total_energy from URL'''
+	total_energy = re.match('^/(?P<device_id>\d+)/total_energy$', path)
```

Comment:
Why not use a parameter for this option? That would be more consistent with our existing style.